libjcat/jcat-engine-gpg.c

```diff
@@ -22,14 +22,20 @@ G_DEFINE_TYPE (JcatEngineGpg, jcat_engine_gpg, JCAT_TYPE_ENGINE)
 G_DEFINE_AUTO_CLEANUP_FREE_FUNC(gpgme_data_t, gpgme_data_release, NULL)
 
 static gboolean
-jcat_engine_gpg_add_public_key (JcatEngineGpg *self,
-			       const gchar *filename,
-			       GError **error)
+jcat_engine_gpg_add_public_key (JcatEngine *engine,
+			        const gchar *filename,
+			        GError **error)
 {
+	JcatEngineGpg *self = JCAT_ENGINE_GPG (engine);
 	gpgme_error_t rc;
 	gpgme_import_result_t result;
 	gpgme_import_status_t s;
 	g_auto(gpgme_data_t) data = NULL;
+	g_autofree gchar *basename = g_path_get_basename (filename);
+
+	/* not us */
+	if (!g_str_has_prefix (basename, "GPG-KEY-"))
+		return TRUE;
```

Comment:
I think if you're doing magic looking at filenames, you should probably output to debug log when keys _aren't_ added too.

Reply:
That sounds better, yeah.

Reply:
Hmm, this doesn't work as we do the per-engine setup now, the context just holds an array and we setup the engines deliberately just when required. I did the g_debug thing like a coward.